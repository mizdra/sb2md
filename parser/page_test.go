package parser

import (
	"reflect"
	"testing"
)

func TestParsePage(t *testing.T) {
	type args struct {
		page_text string
	}
	tests := []struct {
		name string
		args args
		want Page
	}{
		{
			name: "basic",
			args: args{page_text: "title\nbody1\nbody2"},
			want: Page{title: "title", lines: []Line{{text: "title"}, {text: "body1"}, {text: "body2"}}},
		},
		{
			name: "trailing newline",
			args: args{page_text: "title\nbody1\n"},
			want: Page{title: "title", lines: []Line{{text: "title"}, {text: "body1"}, {text: ""}}},
		},
		{
			name: "title only",
			args: args{page_text: "title"},
			want: Page{title: "title", lines: []Line{{text: "title"}}},
		},
		{
			name: "empty title",
			args: args{page_text: "\nbody1"},
			want: Page{title: "", lines: []Line{{text: ""}, {text: "body1"}}},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := ParsePage(tt.args.page_text); !reflect.DeepEqual(got, tt.want) {
				t.Errorf("ParsePage() = %v, want %v", got, tt.want)
			}
		})
	}
}
