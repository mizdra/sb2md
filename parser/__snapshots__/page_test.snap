
[TestExample - 1]
Hello World
---
